app/src/main/java/uk/gov/onelogin/login/nonce/NonceGeneratorModule.kt

```diff
@@ -0,0 +1,16 @@
+package uk.gov.onelogin.login.nonce
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object NonceGeneratorModule {
+
+    @Provides
+    @Singleton
+    fun bindsNonceGenerator(nonceGenerator: NonceGenerator): INonceGenerator = nonceGenerator
+}
```

Comment:
If the implementations are injectable, as shown by having access to the `NonceGenerator` in the function parameters, we're able to change this object into an interface, along with the value. 

Which style do you prefer with your hilt modules?

```suggestion
@Module
@InstallIn(SingletonComponent::class)
interface NonceGeneratorModule {

    @Binds
    @Singleton
    fun bindsNonceGenerator(nonceGenerator: NonceGenerator): INonceGenerator
}
```

Reply:
In jira ticket